src/abstractdataframe/abstractdataframe.jl

```diff
@@ -930,7 +940,22 @@ Base.filter((cols, f)::Pair{<:AbstractVector{Symbol}}, df::AbstractDataFrame) =
 Base.filter((cols, f)::Pair, df::AbstractDataFrame) =
     filter(index(df)[cols] => f, df)
 
-_filter_helper(df, f, cols...) = df[((x...) -> f(x...)::Bool).(cols...), :]
+function _filter_helper(df, f, cols...)
+    if length(cols) == 0
+        throw(ArgumentError("At least one column must be passed to filter on"))
+    end
+    return df[((x...) -> f(x...)::Bool).(cols...), :]
+end
+
+function Base.filter((cols, f)::Pair{<:AsTable}, df::AbstractDataFrame)
+    dff = select(df, cols.colselector, copycols=false)
+    if ncol(dff) == 0
+        throw(ArgumentError("At least one column must be passed to filter on"))
+    end
+    return _filter_helper_astable(df, Tables.namedtupleiterator(dff), f)
+end
+
+_filter_helper_astable(df, nti, f) = df[(x -> f(x)::Bool).(nti), :]
```

Comment:
Define this above?
```suggestion
_filter_helper_astable(df::AbstractDataFrame, nti::NamedTuple, f) = df[(x -> f(x)::Bool).(nti), :]
```

Reply:
OK - I will move it and make the type restriction more strict.